libs/core/algorithms/include/hpx/parallel/algorithms/equal.hpp

```diff
@@ -389,19 +367,12 @@ namespace hpx { namespace parallel { inline namespace v1 {
 
                 typedef hpx::util::zip_iterator<FwdIter1, FwdIter2>
                     zip_iterator;
-                typedef typename zip_iterator::reference reference;
 
                 util::cancellation_token<> tok;
                 auto f1 = [f, tok](zip_iterator it,
                               std::size_t part_count) mutable -> bool {
-                    util::loop_n<std::decay_t<ExPolicy>>(it, part_count, tok,
-                        [&f, &tok](zip_iterator const& curr) mutable -> void {
-                            reference t = *curr;
-                            if (!HPX_INVOKE(f, hpx::get<0>(t), hpx::get<1>(t)))
-                            {
-                                tok.cancel();
-                            }
-                        });
+                    sequential_equal<std::decay_t<ExPolicy>>(
```

Comment:
Here, this lambda is now gets accessed by tag dispatch by `sequential_equal CPO`